### PR TITLE
Bug/654/disable adding pattern when search is empty

### LINKED ISSUE #654: Disable adding pattern when search is empty
# Bug

## Expected Behavior
**When** the search pattern is empty
**then** the add-pattern-menu should disable the appropriate Hide/Exclude option

## Actual Behavior
**When** the search pattern is empty
**then** all options (Hide/Exclude) are available to click inside the add-pattern-menu

**When** the search pattern is already inside the Hide/Exclude list
**then** the add-pattern-menu disables the appropriate Hide/Exclude option

## Comments
The `searchBar.component.html` still contains some dead code, which was used before to disable the options depending on the amount of matching files/folders from the search.

## Question
Should we also disable the options, when no nodes match the pattern, like before or only when the pattern is empty?

## Steps to Reproduce the Problem

  1. Open codeCharta
  1. click the "three dot"-button to add a blacklist pattern
  1. click the option Hide or Exclude
  1. Now an empty pattern was added to the blacklist

## Specifications

  - Version: 1.30.0
  - Platform: macOS
  - Browser: Safari

### PR DESCRIPTION
**Pull request by JaiJoPa**
# Disable adding pattern when search is empty

closes #654 

## Description

When the search pattern is empty, the options to exclude or hide the empty string is disabled. 

empty_string

If the search pattern is already in the exclude or hide list, then that option is disabled. 

already_excluded

If the pattern won't match any building, the options are not disabled.

no_matching

## Definition of Done

A task/pull request will not be considered to be complete until all these items can be checked off.

- [x] **All** requirements mentioned in the issue are implemented
- [x] Does match the Code of Conduct and the Contribution file
- [x] Task has its own **GitHub issue** (something it is solving)
  - [x] Issue number is **included in the commit messages** for traceability
- [x] **Update the README.md** with any changes/additions made
- [x] **Update the CHANGELOG.md** with any changes/additions made
- [x] **Enough test coverage to ensure that uncovered changes do not break functionality**
- [x] **All tests pass**
- [x] **Descriptive pull request text**, answering:
  - What problem/issue are you fixing?
  - What does this PR implement and how?
- [x] **Assign your PR to someone** for a code review
  - This person _will be contacted **first**_ if a bug is introduced into `master`
- [x] **Manual testing** did not fail

### CODE REVIEW

File: visualization/app/codeCharta/ui/searchBar/searchBar.component.ts [hunk 1]
```diff
@@ -46,6 +46,14 @@ export class SearchBarController implements BlacklistSubscriber {
 		this.resetSearchPattern()
 	}
 
+	public isSearchPatternEmpty() {
+		return this._viewModel.searchPattern === ""
+	}
+
+	public isPatternInBlackList() {
+		this.updateViewModel(this.settingsService.getSettings().fileSettings.blacklist)
```
Comment (alschmut) [inline visualization/app/codeCharta/ui/searchBar/searchBar.component.ts]: the function name does not fit to what it does!

Comment (alschmut) [reply visualization/app/codeCharta/ui/searchBar/searchBar.component.ts]: And why do we need this function? whenever the blacklist changes we update our viewModel..

Comment (alschmut) [reply visualization/app/codeCharta/ui/searchBar/searchBar.component.ts]: class was refactored.

**Comment by JaiJoPa** [comment]
@alschmut , it looks good :)

**Pull request merged**